MoreLinq/Create.cs

```diff
@@ -0,0 +1,48 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2017 Atif Aziz. All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+
+    static partial class MoreEnumerable
+    {
+        /// <summary>
+        /// Creates an <see cref="IEnumerable{T}"/> given a factory function
+        /// for an <see cref="IEnumerator{T}"/>.
+        /// </summary>
+        /// <typeparam name="T">Type of returned sequence elements.</typeparam>
+        /// <param name="factory">
+        /// Function that creates an <see cref="IEnumerator{T}"/>.</param>
+        /// <remarks>
+        /// This method uses deferred execution semantics and streams its
+        /// results.
+        /// </remarks>
+
+        public static IEnumerable<T> Create<T>(Func<IEnumerator<T>> factory)
```

Comment:
`Wrap` makes sense when you're not changing the type.

> After all, most methods in MoreLinq create an `IEnumerable<T>`

Yeah and most of them are named after the operation they perform and so I would have wanted to call it `Enumerable` (because it turns an `IEnumerator<>` returning function into an `IEnumerable<>`) but `MoreEnumerable.Enumerable` seemed ugly. I also thought about `Delegate` because it returns an `IEnumerable<T>` that merely delegates its implementation to a function. However, it's really the most basic _creation function_ you can possibly imagine/implement and so I'd say `Create` sounds about right. Oh and look, [System.Interactive has one going by the same name](https://github.com/Reactive-Extensions/Rx.NET/blob/f47d85ef7cb70f6ba9eecc5f82462a13f3b8e821/Ix.NET/Source/System.Interactive/Create.cs). 😄